### PR TITLE
Add 3d bomex box

### PR DESCRIPTION
**Pull request by charleskawczynski**
Similar to #810, this PR adds a 3D box of Bomex, that calls TC.jl (albeit using the wrong space). This may be a simpler stepping stone to get working before moving to the sphere since viz and transformations will be simpler/easier.

### CODE REVIEW

File: integration_tests/3dBomex.jl [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    cp_d = C_p
+    cv_d = C_v
+    p_0 = MSLP
+    g = grav
+
+    # auxiliary quantities
+    r = sqrt((x - x_c)^2 + (z - z_c)^2)
+    θ_p = r < r_c ? 0.5 * θ_c * (1.0 + cospi(r / r_c)) : 0.0 # potential temperature perturbation
+
+    θ = θ_b + θ_p # potential temperature
```
Comment (yairchn) [inline integration_tests/3dBomex.jl:100]: here we need to load the initial condition of the Bomex case

Comment (charleskawczynski) [reply integration_tests/3dBomex.jl:102]: Completely agree, let's do this in a follow-up PR.

File: integration_tests/3dBomex.jl [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+    π_exn = 1.0 - g * z / cp_d / θ # exner function
+    T = π_exn * θ # temperature
+    p = p_0 * π_exn^(cp_d / R_d) # pressure
+    ρ = p / R_d / T # density
+    ρθ = ρ * θ # potential temperature density
+
+    # Horizontal momentum defined on cell centers
+    return (ρ = ρ, ρθ = ρθ, ρuₕ = ρ * CCG.UVVector(0.0, 0.0))
```
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply integration_tests/3dBomex.jl:110]: Completely agree, let's do this in a follow-up PR.

File: integration_tests/3dBomex.jl [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+
+    #  1c) horizontal div of horizontal grad of vert momentum
+    @. dρw += hdiv(κ₂ * (Yfρ * hgrad(ρw / Yfρ)))
+    #  1d) vertical div of vertical grad of vert momentun
+    @. dρw += vvdivc2f(κ₂ * (Yc.ρ * ∂c(ρw / Yfρ)))
+
+    #  2a) horizontal div of horizontal grad of potential temperature
+    @. dρθ += hdiv(κ₂ * (Yc.ρ * hgrad(ρθ / ρ)))
+    #  2b) vertical div of vertial grad of potential temperature
+    @. dρθ += ∂(κ₂ * (Yfρ * ∂f(ρθ / ρ)))
```
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply integration_tests/3dBomex.jl:313]: Completely agree, let's do this in a follow-up PR.

**Comment by charleskawczynski** [comment]
Alright, this is running, many changes are needed, but let's merge so that more people can work on it in parallel.

**Pull request merged**